tracer/src/Datadog.Trace/Debugger/Upload/SymbolUploadApi.cs

```diff
@@ -84,24 +84,24 @@ public override async Task<bool> SendBatchAsync(ArraySegment<byte> symbols)
 
             MultipartFormItem symbolsItem;
 
-            if (_enableCompression)
+            if (!this._enableCompression)
+            {
+                symbolsItem = new MultipartFormItem("file", MimeTypes.Json, "file.json", symbols);
+            }
+            else
             {
                 using var memoryStream = new MemoryStream();
 #if NETFRAMEWORK
-                using var gzipStream = new Vendors.ICSharpCode.SharpZipLib.GZip.GZipOutputStream(memoryStream);
-                await gzipStream.WriteAsync(symbols.Array, 0, symbols.Array.Length).ConfigureAwait(false);
-                await gzipStream.FlushAsync().ConfigureAwait(false);
+                using (var gzipStream = new Vendors.ICSharpCode.SharpZipLib.GZip.GZipOutputStream(memoryStream))
 #else
-                using var gzipStream = new GZipStream(memoryStream, CompressionMode.Compress);
-                await gzipStream.WriteAsync(symbols.Array, 0, symbols.Array.Length).ConfigureAwait(false);
-                await gzipStream.FlushAsync().ConfigureAwait(false);
+                using (var gzipStream = new GZipStream(memoryStream, CompressionMode.Compress))
 #endif
+                {
+                    await gzipStream.WriteAsync(symbols.Array, 0, symbols.Array.Length).ConfigureAwait(false);
```

Comment:
It is best to _also_ call `FlushAsync()` here even though it isn't _strictly_ required, because it avoids a sync-over-async case that can happen in the `Dispose()` call otherwise:

```suggestion
                    await gzipStream.WriteAsync(symbols.Array, 0, symbols.Array.Length).ConfigureAwait(false);
                    await gzipStream.FlushAsync().ConfigureAwait(false);
```

The other solution is to use `await using`, but that's only available in later TFMs, so the explicit flush is the better option here 🙂